ayushma/views/document.py

```diff
@@ -41,3 +44,11 @@ def perform_create(self, serializer):
         document = serializer.save(project=project)
         # upsert file
         upsert(filepath=str(document.file), external_id=external_id)
+
+    '''If an admin or a superuser deletes a document, it's vectors are deleted from the pinecone database'''
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        self.perform_destroy(instance)
+        if request.user.is_staff or int(request.data['user_type'] == 3):
```

Comment:
do you mean `self.request.user.user_type`?

Reply:
same thing :P